src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public a: number,
+    public b: number,
+    public c: number,
+    public color: Color,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    const longestSide = Math.max(this.a, this.b, this.c);
+    const sumOfOther = a + b + c - longestSide;
+
+    if (longestSide > sumOfOther) {
+      throw new Error('Longest side > than sum of other sides');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return parseFloat(area.toFixed(2));
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public radius: number,
+    public color: Color,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = this.radius * this.radius * Math.PI;
+
+    return parseFloat(area.toFixed(2));
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public width: number,
+    public height: number,
+    public color: Color,
+  ) {
+    if (this.width <= 0 || this.height <= 0) {
+      throw new Error('Width and height must be greater than 0');
+    }
+  }
+
+  getArea(): number {
+    const area = this.width * this.height;
+
+    return parseFloat(area.toFixed(2));
+  }
+}
 
-export function getInfo(figure): string {
-  return typeof figure;
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
```

Comment:
The 'getInfo' function does not format the output string according to the task description. The task requires the format to be 'A {color} {shape} with area {area}', but the current implementation uses a hyphen instead of 'with area'. Please update the string format to match the task requirements.